core/src/service/drawing_service.rs

```diff
@@ -0,0 +1,351 @@
+            ),
+            SupportedImageFormats::Pnm => PnmEncoder::new(&mut buf_writer).encode(
+                drawing_bytes.as_slice(),
+                width,
+                height,
+                ColorType::Rgba8,
+            ),
+            SupportedImageFormats::Jpeg => JpegEncoder::new(&mut buf_writer).encode(
+                drawing_bytes.as_slice(),
+                width,
+                height,
+                ColorType::Rgba8,
+            ),
+            SupportedImageFormats::Tga => TgaEncoder::new(&mut buf_writer).encode(
+                drawing_bytes.as_slice(),
+                width,
+                height,
+                ColorType::Rgba8,
+            ),
+            SupportedImageFormats::Farbfeld => FarbfeldEncoder::new(&mut buf_writer).encode(
+                drawing_bytes.as_slice(),
+                width,
+                height,
+            ),
+            SupportedImageFormats::Bmp => BmpEncoder::new(&mut buf_writer).encode(
+                drawing_bytes.as_slice(),
+                width,
+                height,
+                ColorType::Rgba8,
+            ),
+        }
+        .map_err(DrawingError::FailedToEncodeImage)?;
+
+        std::mem::drop(buf_writer);
+
+        Ok(buffer)
+    }
+}
+
+impl<
+        MyBackend: Backend,
+        MyFileService: FileService<MyBackend>,
+        FileMetadataDb: FileMetadataRepo<MyBackend>,
+    > DrawingServiceImpl<MyBackend, MyFileService, FileMetadataDb>
+{
+    fn u32_byte_to_u8_bytes(u32_byte: u32) -> (u8, u8, u8, u8) {
+        let mut byte_1 = (u32_byte >> 16) & 0xffu32;
+        let mut byte_2 = (u32_byte >> 8) & 0xffu32;
+        let mut byte_3 = u32_byte & 0xffu32;
+        let byte_4 = (u32_byte >> 24) & 0xffu32;
+
+        if byte_4 > 0u32 {
+            byte_1 = byte_1 * 255u32 / byte_4;
+            byte_2 = byte_2 * 255u32 / byte_4;
+            byte_3 = byte_3 * 255u32 / byte_4;
+        }
+
+        (byte_1 as u8, byte_2 as u8, byte_3 as u8, byte_4 as u8)
+    }
+
+    fn get_drawing_bounds(strokes: &[Stroke]) -> Result<(u32, u32), DrawingError<MyBackend>> {
+        let mut greatest_width = 1;
+        let mut greatest_height = 1;
+
+        for stroke in strokes {
+            if stroke.points_x.len() != stroke.points_y.len()
+                || stroke.points_y.len() != stroke.points_girth.len()
+            {
+                return Err(DrawingError::UnequalPointsAndGirthMetrics);
+            }
+
+            for point_index in 0..stroke.points_x.len() - 1 {
+                let point_girth_1 = stroke
+                    .points_girth
+                    .get(point_index)
+                    .ok_or(DrawingError::CorruptedDrawing)?
+                    .to_owned() as u32;
+                let point_girth_2 = stroke
+                    .points_girth
+                    .get(point_index + 1)
+                    .ok_or(DrawingError::CorruptedDrawing)?
+                    .to_owned() as u32;
+                let x1 = stroke
```

Comment:
It's not clear to me why you're not just going through all the points and finding the max X and max Y. Why are you doing this x1, x2 stuff?